### PR TITLE
fix: get tx receipt should return even if reverted tx

### PR DESCRIPTION
**Pull request by 0xArdi**
## Fixes

Get tx receipt should return the receipt even if the tx has reverted.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Adamantios** [comment]
We already have a WIP in #772 for bumping macos version which fixes one of two issues that surface. I would suggest you to ignore the mac tests until this is merged.

File: packages/valory/connections/test_libp2p/tests/base.py
Comment (Adamantios) [inline packages/valory/connections/test_libp2p/tests/base.py:1]: Is this required?

Comment (0xArdi) [reply packages/valory/connections/test_libp2p/tests/base.py:1]: On macos-14 it complained that cwd doesn't exist.

File: packages/valory/connections/ledger/tests/test_ledger_api.py [hunk 1]
```diff
@@ -638,7 +638,7 @@ async def test_no_raw_tx(
     @pytest.mark.asyncio
     @pytest.mark.parametrize(
         "failing_ledger_method_name",
-        ("get_transaction_receipt", "is_transaction_settled", "get_transaction"),
+        ("get_transaction_receipt", "get_transaction"),
```
Comment (Adamantios) [inline packages/valory/connections/ledger/tests/test_ledger_api.py:641]: Some checks below do not make sense anymore, e.g.:
https://github.com/valory-xyz/open-aea/blob/154eb91729a4fbe29feaf3006e8dff641f224738/packages/valory/connections/ledger/tests/test_ledger_api.py#L676

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -433,7 +433,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-12, windows-latest]
+        os: [ubuntu-latest, windows-latest]
```
Comment (Adamantios) [inline .github/workflows/workflow.yml:436]: Why?

File: packages/valory/connections/test_libp2p/tests/base.py
Comment (Adamantios) [reply packages/valory/connections/test_libp2p/tests/base.py:1]: Can you point to this from the CI?

File: .github/workflows/workflow.yml (hunk 2 above)
Comment (0xArdi) [reply .github/workflows/workflow.yml:436]: To skip platform checks for now.

Comment (Adamantios) [reply .github/workflows/workflow.yml:436]: Why not skip only the specific test that fails?

File: packages/valory/connections/test_libp2p/tests/base.py
Comment (0xArdi) [reply packages/valory/connections/test_libp2p/tests/base.py:1]: https://github.com/valory-xyz/open-aea/actions/runs/12263381150/job/34214989157#step:9:13811

Comment (Adamantios) [reply packages/valory/connections/test_libp2p/tests/base.py:1]: Ok but this looks like a side effect of the failure before that, and this change either way causes [FileNotFoundError: [Errno 2] No such file or directory: ''](https://github.com/valory-xyz/open-aea/actions/runs/12278181212/job/34259522443#step:9:13874)

**Pull request merged**